e2e/alertmanager_test.go

```diff
@@ -20,188 +20,172 @@ import (
+	ctx := context.Background()
+	kubeClient, opClient, err := newKubeClients()
+	if err != nil {
+		t.Fatalf("error instantiating clients. err: %s", err)
+	}
+
+	t.Run("alertmanager-deployed", testAlertmanagerDeployed(ctx, t, kubeClient))
+	t.Run("alertmanager-configured", testAlertmanagerConfigured(ctx, t, kubeClient, opClient))
+}
+
+func testAlertmanagerDeployed(ctx context.Context, t *testing.T, kubeClient kubernetes.Interface) func(*testing.T) {
+	return func(t *testing.T) {
+		t.Log("checking alertmanager is running")
+
+		err := wait.PollUntilContextCancel(ctx, pollDuration, false, func(ctx context.Context) (bool, error) {
+			ss, err := kubeClient.AppsV1().StatefulSets(operator.DefaultOperatorNamespace).Get(ctx, operator.NameAlertmanager, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			} else if err != nil {
+				return false, fmt.Errorf("getting alertmanager StatefulSet failed: %w", err)
+			}
+
+			// Ensure alertmanager pod is ready.
+			if ss.Status.ReadyReplicas != 1 {
+				return false, nil
+			}
+
+			// Assert we have the expected annotations.
+			wantedAnnotations := map[string]string{
+				"components.gke.io/component-name":               "managed_prometheus",
+				"cluster-autoscaler.kubernetes.io/safe-to-evict": "true",
+			}
+			if diff := cmp.Diff(wantedAnnotations, ss.Spec.Template.Annotations); diff != "" {
+				return false, fmt.Errorf("unexpected annotations (-want, +got): %s", diff)
+			}
+
+			// Ensure default no-op alertmanager secret has been created by operator.
+			secret, err := kubeClient.CoreV1().Secrets(operator.DefaultOperatorNamespace).Get(ctx, operator.AlertmanagerSecretName, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			} else if err != nil {
+				return false, fmt.Errorf("getting alertmanager StatefulSet failed: %w", err)
+			}
 
-	alertmanagerConfig := `
+			bytes, ok := secret.Data[operator.AlertmanagerConfigKey]
+			if !ok {
+				return false, errors.New("getting alertmanager secret data failed")
+			}
+
+			if diff := cmp.Diff([]byte(operator.AlertmanagerNoOpConfig), bytes); diff != "" {
+				return false, fmt.Errorf("unexpected secret payload (-want, +got): %s", diff)
+			}
+
+			return true, nil
+		})
+		if err != nil {
+			t.Fatalf("waiting for alertmanager Statefulset failed: %s", err)
+		}
+	}
+}
+
+func testAlertmanagerConfigured(ctx context.Context, t *testing.T, kubeClient kubernetes.Interface, opClient versioned.Interface) func(*testing.T) {
+	return func(t *testing.T) {
+		t.Log("checking alertmanager is configured")
+
+		// Provision custom Alertmanager secret.
+		alertmanagerConfig := `
 receivers:
   - name: "foobar"
 route:
   receiver: "foobar"
 `
-	testAlertmanager(context.Background(), tctx, nil,
-		&corev1.Secret{
+		secret := &corev1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      operator.AlertmanagerPublicSecretName,
-				Namespace: tctx.pubNamespace,
+				Name: "my-secret-name",
```

Comment:
question: What is the advantage of the literal here, over the constant `operator.AlertmanagerPublicSecretName`?